### PR TITLE
Xml parser strict

### PR DESCRIPTION
**Pull request by armstrmi**
## Description of Changes

- added 'strict' feature to the xml-parser 
- updated docs under xml-parser.md
- updated tests to include edge case for special characters in event of strict = false

## **Please check that the PR fulfills these requirements**
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)
- [ ] Add a changelog entry (for non-trivial bug fixes / features)
- [ ] CI passes

### CODE REVIEW

File: operator/builtin/parser/xml/xml_test.go [hunk 1]
```diff
@@ -12,9 +12,12 @@ import (
 )
 
 func TestParse(t *testing.T) {
+	var disabled bool = false
```
Comment (dehaansa) [inline operator/builtin/parser/xml/xml_test.go:15]: Should be a little clearer - strictDisabled probably

**Review by dehaansa — APPROVED** [review]
